lib/Conversion/TorchOnnxToTorch/OnnxLstmExpander.cpp

```diff
@@ -0,0 +1,479 @@
+  Value HSizeX4 = rewriter.create<Torch::ConstantIntOp>(
+      loc, intType,
+      rewriter.getIntegerAttr(rewriter.getIntegerType(64), 4 * hidden_size));
+
+  Torch::ValueTensorType GateWeightsTypeIH =
+      rewriter.getType<Torch::ValueTensorType>(
+          llvm::SmallVector<int64_t>{hidden_size, input_size},
+          W.getType().cast<Torch::ValueTensorType>().getDtype());
+  Value W_i = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateWeightsTypeIH, W, cstZero, cstZero, HSizeX1, cstOne);
+  Value W_o = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateWeightsTypeIH, W, cstZero, HSizeX1, HSizeX2, cstOne);
+  Value W_f = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateWeightsTypeIH, W, cstZero, HSizeX2, HSizeX3, cstOne);
+  Value W_c = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateWeightsTypeIH, W, cstZero, HSizeX3, HSizeX4, cstOne);
+
+  Torch::ValueTensorType GateWeightsTypeHH =
+      rewriter.getType<Torch::ValueTensorType>(
+          llvm::SmallVector<int64_t>{hidden_size, hidden_size},
+          R.getType().cast<Torch::ValueTensorType>().getDtype());
+  Value R_i = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateWeightsTypeHH, R, cstZero, cstZero, HSizeX1, cstOne);
+  Value R_o = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateWeightsTypeHH, R, cstZero, HSizeX1, HSizeX2, cstOne);
+  Value R_f = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateWeightsTypeHH, R, cstZero, HSizeX2, HSizeX3, cstOne);
+  Value R_c = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateWeightsTypeHH, R, cstZero, HSizeX3, HSizeX4, cstOne);
+
+  Torch::ValueTensorType GateBiasType =
+      rewriter.getType<Torch::ValueTensorType>(
+          llvm::SmallVector<int64_t>{hidden_size},
+          Wb.getType().cast<Torch::ValueTensorType>().getDtype());
+  Value Wb_i = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateBiasType, Wb, cstZero, cstZero, HSizeX1, cstOne);
+  Value Wb_o = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateBiasType, Wb, cstZero, HSizeX1, HSizeX2, cstOne);
+  Value Wb_f = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateBiasType, Wb, cstZero, HSizeX2, HSizeX3, cstOne);
+  Value Wb_c = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateBiasType, Wb, cstZero, HSizeX3, HSizeX4, cstOne);
+  Value Rb_i = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateBiasType, Rb, cstZero, cstZero, HSizeX1, cstOne);
+  Value Rb_o = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateBiasType, Rb, cstZero, HSizeX1, HSizeX2, cstOne);
+  Value Rb_f = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateBiasType, Rb, cstZero, HSizeX2, HSizeX3, cstOne);
+  Value Rb_c = rewriter.create<Torch::AtenSliceTensorOp>(
+      loc, GateBiasType, Rb, cstZero, HSizeX3, HSizeX4, cstOne);
+
+  auto YType = rewriter.getType<Torch::ValueTensorType>(
+      ArrayRef<int64_t>{seq_len, batch_size, hidden_size}, HType.getDtype());
```

Comment:
You can't make an array ref from an initializer list as this is not memory safe. Make a `SmallVector` and pass it in.

Reply:
fixed